internal/provider/resources/block.go

```diff
@@ -6,6 +6,7 @@ import (
 	"fmt"
 	"strings"
 
+	"github.com/avast/retry-go/v4"
```

Comment:
adding this retry library, it's a wrapper that has some nice, flexible, low commitment interfaces